improve select and populate query variables

ehmicky opened the pull request:
At the moment, using `?select=field.subfield` does not work, nor does `?populate=field.subfield`.
This commit aims at providing this feature. However, I simplified the code quite a lot, so I hope I am not missing anything important here.
The build is failing at the moment (jshint + test units), but I would like to know your feedback, since it seems like this feature has already been implemented (according to the test unit), so this pull request might be useless.

florianholzapfel commented:
this one should work ```?populate=field&select=field.subfield``` (it fetches all items from the current model, populates ```field``` and selects only ```field.subfield```).

ehmicky commented:
Sorry, this does work as expect. My problem with `?populate=field.subfield` is about recursive population, which does not seem to be implemented in Mongoose right now (you need to do two separate Mongoose queries). This could be an interesting feature for express-restify-mongoose.

ehmicky commented:
Sorry to reopen this but I actually think there is a problem when it comes to selecting subfield, e.g. using `select=field.subfield`. It just does not do anything for me.

The piece of code that handles it is the following:

    if (queryOptions.current.select) {
        arr = queryOptions.current.select.split(',');
        for (i = 0; i < arr.length; ++i) {
            var selectItem = arr[i];
            var selectionModifier = 1;
            if (selectItem.match(/^\-/)) { // exclusion
                selectItem = selectItem.substring(1);
                selectionModifier = 0;
            }
            if (selectItem.match(/\./)) {
                var subSelect = selectItem.split('.');
                if (!selectObj[subSelect[0]]) {
                    selectObj[subSelect[0]] = {};
                }
                selectObj[subSelect[0]][subSelect[1]] = selectionModifier;
            } else {
                selectObj.root[selectItem] = selectionModifier;
            }
        }
        query = query.select(selectObj.root);
    }

`subSelect` would be `["field", "subfield"]` and selectObj `{root: {}, countries: {year: 1}}`, which would end up calling `query.select({})` which does nothing. As a result querying a subfield does not work.
